app/src/main/java/org/openobservatory/ooniprobe/adapters/ResultListAdapter.java

```diff
@@ -0,0 +1,298 @@
+                        DrawableCompat.setTint(DrawableCompat.wrap(cBinding.binding.failedMeasurements.getCompoundDrawablesRelative()[0]).mutate(), ContextCompat.getColor(cBinding.binding.failedMeasurements.getContext(), blocked == 0 ? R.color.color_gray9 : R.color.color_yellow9));
+                        boolean allUploaded = true;
+                        for (Measurement m : result.getMeasurements())
+                            allUploaded = allUploaded && (m.isUploaded() || m.is_failed);
+                        cBinding.binding.startTime.setCompoundDrawablesRelativeWithIntrinsicBounds(0, 0, allUploaded ? 0 : R.drawable.cloudoff, 0);
+
+                    }
+                    break;
+                    case 5:
+                    default: {
+                        ViewHolder<ItemExperimentalBinding> defaultHolder = (ViewHolder<ItemExperimentalBinding>) holder;
+                        defaultHolder.binding.asnName.setText(Network.toString(defaultHolder.binding.asnName.getContext(), result.network));
+                        defaultHolder.binding.startTime.setText(DateFormat.format(DateFormat.getBestDateTimePattern(Locale.getDefault(), "yMdHm"), result.start_time));
+                        boolean allUploaded = true;
+                        for (Measurement m : result.getMeasurements())
+                            allUploaded = allUploaded && (m.isUploaded() || m.is_failed);
+                        defaultHolder.binding.startTime.setCompoundDrawablesRelativeWithIntrinsicBounds(0, 0, allUploaded ? 0 : R.drawable.cloudoff, 0);
+                    }
+                    break;
+                }
+            } else if (iResult instanceof UiModel.SeparatorModel) {
+                ViewHolder<ItemDateBinding> vHolder = (ViewHolder<ItemDateBinding>) holder;
+                vHolder.binding.textView.setText(((UiModel.SeparatorModel) iResult).mDescription);
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public int getItemViewType(int position) {
+        UiModel iResult = Objects.requireNonNull(getItem(position));
+        if (iResult instanceof UiModel.SeparatorModel) {
+            return -2;
+        }
+        Result result = ((UiModel.ResultModel) iResult).item;
+        if (result.countTotalMeasurements() == 0) {
+            return -1;
+        }
+        switch (result.test_group_name) {
+            case WebsitesSuite.NAME:
+                return 0;
+            case InstantMessagingSuite.NAME:
+                return 1;
+            case MiddleBoxesSuite.NAME:
+                return 2;
+            case PerformanceSuite.NAME:
+                return 3;
+            case CircumventionSuite.NAME:
+                return 4;
+            case ExperimentalSuite.NAME:
+            default:
+                return 5;
```

Comment:
These could be defined as ENUMs or const, so that you aren't doing switch casing on opaque numbers.